portals/knowledge-management/src/Pages/Home.vue

```diff
@@ -87,11 +90,25 @@
             searchResults.value = response.data;
             loadingResults.value = false;
         });
+        globalSearchQuery.value = '';
     }, 500);
 
     const { user } = useAuthStore();
     const { hasServiceManagement } = useFeatureStore();
 
+    onMounted(function () {
+        if (route.query.search !== undefined) {
+            searchQuery.value = route.query.search;
+        }
+    });
+
+    watch(
+        () => route.query.search,
+        (newVal, oldVal) => {
+            searchQuery.value = newVal || '';
+        },
+    );
```

Comment:
Do we need the `onMounted` call here? I feel that this may cause the same issue we had with the page view count where this is triggered twice.

If the reason is so that you can clear the search input, I think you should be fine doing that locally within the component after the route push

Reply:
@Orrison No It would never trigger twice as onMounted only run once when the component is rendered while watch runs when the value in this case 'route.query.search' changes.

Reply:
@jayushi-canyon got it. Though it may not be triggered twice. I am pretty sure the `onMounted` is redundant here, so it would be best to remove it.